### ISSUE TITLE
Testing: Using `WebElement#Click` instead of `Click()`

### ISSUE THREAD

**Issue by crphang** [TEAMMATES/teammates#8623]
This topic was discussed in #7886 and some additional changes was made in #7887 to change to Javascript Click. In those discussions, it was raised that `WebElement#Click` does not work sometimes even with maximizing the browser. However, it is still true that `WebElement#Click` models a real click action better, and Javascript Click can click the element even if it is not visible to the user.

#8373 discusses this further as tests fail on local development environment for `WebElement#Click` but passes using Javascript Click. Further investigations together with @LiHaoTan prompted that the page have yet to scrolled, and the test already attempted to click the element. Meaning we need to wait some time for the page to scroll before clicking. Adding `ThreadHelper.wait(1000)` to wait 1s for the page to scroll allow the test to pass with `WebElement#Click`.

Even though `WebElement#Click` fails to work reliably but it might be because we aren't careful in it's usage. Adding `ThreadHelper.wait(1000)` might allow replacement of other `click()` to `WebElement#Click` to work reliably as well.

It remains necessary to discuss whether waiting 1s per click adds too much testing overhead and other possibilities that `WebElement#Click` does not work well.

**Comment by whipermr5** [comment]
Scrolling is slow... I expect much longer build times if we were to do this.

**Comment by LiHaoTan** [comment]
Yeah, it is slow but I guess it is better that UI tests are reliable (since Javascript click has a number of caveats such as being able to magically click)

I guess maybe this will also force us to try to avoid writing UI tests unless there is a need to.

**Comment by damithc** [comment]
Also depends on how big the slow down is. Sometimes we think it's just a few seconds but can turn out 10 extra minutes! :-p
I suppose we can also use a mix. i.e., use the fast method most of the time but use the reliable but slow method only when needed.

**Comment by damithc** [comment]
To be fair, even an extra 10 minutes may be worth it if it passes all the tests in one go, because the extra time needed for a retry can be significant too.

**Comment by whipermr5** [comment]
This is not about reducing instability, but reliability (false positives). @LiHaoTan's point is that UI tests can pass by clicking elements, even elements that are hidden and can't be clicked. However, I feel that up to this point the UI tests have been quite reliable.

**Comment by Crphang** [comment]
@LiHaoTan's idea is that it gives us more confidence that the test is correct and that it is a proper UI test. Using Javascript Click puts more responsibility on the developer to know that it is working as intended. The workaround that was found using `ThreadHelper.wait()` should allow us to reliably change Javascript Clicks to WebElement#Click, if the Javascript Click is working as intended.

But do you guys people think its a necessary change to ensure that all future ui tests are more reliable?

**Comment by whipermr5** [comment]
I feel waiting is not ideal. What if we change the screen resolution to something really huge so that scrolling is not necessary?
https://github.com/TEAMMATES/teammates/blob/4d8b09070cf9ae55b45fee493896704a5877a082/.travis.yml#L28-L29

This won't work for testing locally though, unless we go headless like in CI.

**Comment by LiHaoTan** [comment]
The scrolling behavior didn't come from testing itself (it came from the way the page functions). Not waiting for the scroll to complete is what created the surprise.

I think for UI tests we should never go into such workarounds unless there is a need, there's already enough flakiness due to the complexity of UI tests and I don't think adding more to it is a good idea.

Other surprises are when we have buttons that are theoretically visible but blocked by navbars.

Of course, depending on an arbitrary wait time is not ideal but we can always fix that by waiting for the scrolling to stop.

**Comment by wkurniawan07** [comment]
Closing as this is not a priority right now
